src/main/java/org/kairosdb/auth/PulseAuthModule.java

```diff
@@ -0,0 +1,785 @@
+     * Context.INITIAL_CONTEXT_FACTORY
+     */
+    private String _contextFactory;
+
+    /**
+     * root DN used to connect to
+     */
+    private String _bindDn;
+
+    /**
+     * password used to connect to the root ldap context
+     */
+    private String _bindPassword;
+
+    /**
+     * object class of a user
+     */
+    private String _userObjectClass = "inetOrgPerson";
+
+    /**
+     * attribute that the principal is located
+     */
+    private String _userRdnAttribute = "uid";
+
+    /**
+     * attribute that the principal is located
+     */
+    private String _userIdAttribute = "cn";
+
+    /**
+     * name of the attribute that a users password is stored under
+     * <p>
+     * NOTE: not always accessible, see force binding login
+     */
+    private String _userPasswordAttribute = "userPassword";
+
+    /**
+     * base DN where users are to be searched from
+     */
+    private String _userBaseDn;
+
+    /**
+     * base DN where role membership is to be searched from
+     */
+    private String _roleBaseDn;
+
+    /**
+     * object class of roles
+     */
+    private String _roleObjectClass = "groupOfUniqueNames";
+
+    /**
+     * name of the attribute that a username would be under a role class
+     */
+    private String _roleMemberAttribute = "uniqueMember";
+
+    /**
+     * the name of the attribute that a role would be stored under
+     */
+    private String _roleNameAttribute = "roleName";
+
+    private boolean _debug;
+
+    /**
+     * if the getUserInfo can pull a password off of the user then
+     * password comparison is an option for authn, to force binding
+     * login checks, set this to true
+     */
+    private boolean _forceBindingLogin = false;
+
+    /**
+     * When true changes the protocol to ldaps
+     */
+    private boolean _useLdaps = false;
+
+    /**
+     * Toggles the softAuth configuration of this module. Basically this allows users/services to test their
+     * credentials before authentication is fully enforced. This allows traffic without auth in but if credentials
+     * are provided they are tested against LDAP. Also needs to be set in kairosdb.conf under jetty_auth.
+     */
+    private boolean _softAuth = false;
+
+    private DirContext _rootContext;
+
+
+    public class LDAPUserInfo extends UserInfo
+    {
+        Attributes attributes;
+
+        /**
+         * @param userName the user name
+         * @param credential the credential
+         * @param attributes the user {@link Attributes}
+         */
+        public LDAPUserInfo(String userName, Credential credential, Attributes attributes)
+        {
+            super(userName, credential);
+            this.attributes = attributes;
+        }
+
+        @Override
+        public List<String> doFetchRoles() throws Exception
+        {
+            return getUserRoles(_rootContext, getUserName(), attributes);
+        }
+
+    }
+
+
+    /**
+     * get the available information about the user
```

Comment:
In general I think you should use consistent formatting for your comments. Sometimes you are using capitalization for sentences and sometimes you aren't. I suggest changing all comments to use correct capitalization.

Reply:
Fixed on latest commit